src/file_storage.cpp

```diff
@@ -802,7 +802,7 @@ namespace libtorrent
 	std::int64_t file_storage::file_size(int const index) const
 	{
 		TORRENT_ASSERT_PRECOND(index >= 0 && index < int(m_files.size()));
-		return std::int64_t(m_files[std::size_t(index)].size);
+		return m_files[std::size_t(index)].size;
```

Comment:
about all these casts for operator[]. it feels a bit unfortunate. is it msvc that warns on this otherwise? I seem to recall that ``-Wconversion`` doesn't warn on this